### PR TITLE
[BAEL-1951]create new module core-java-nio and adding java filechannel examples

### PR DESCRIPTION
**Pull request by anuraggoyal1**

### CODE REVIEW

File: core-java-modules/core-java-nio/pom.xml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+		<resources>
+			<resource>
+				<directory>src/main/resources</directory>
+				<filtering>true</filtering>
+			</resource>
+		</resources>
+		<plugins>
+			<plugin>
+				<groupId>org.codehaus.mojo</groupId>
+				<artifactId>exec-maven-plugin</artifactId>
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-nio/pom.xml]: This looks to be trying to run a `com.baeldung.outofmemoryerror.OutOfMemoryGCLimitExceed` class... this plugin shouldn't be needed!

File: core-java-modules/core-java-nio/pom.xml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	</build>
+
+	<profiles>
+		<profile>
+			<id>integration</id>
+			<build>
+				<plugins>
+					<plugin>
+						<groupId>org.codehaus.mojo</groupId>
+						<artifactId>exec-maven-plugin</artifactId>
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-nio/pom.xml]: Is this plugin needed?

Comment (anuraggoyal1) [reply core-java-modules/core-java-nio/pom.xml]: i believe it is not needed. removed it.

File: core-java-modules/core-java-nio/pom.xml (hunk 1 above)
Comment (anuraggoyal1) [reply core-java-modules/core-java-nio/pom.xml]: Yes. removed.

File: core-java-modules/core-java-nio/README.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (KevinGilmore) [inline core-java-modules/core-java-nio/README.md]: For brand new modules, we don't have to create this file up-front. The editor will add this file once the article is published, as the final URL won't be determined until then.

Comment (anuraggoyal1) [reply core-java-modules/core-java-nio/README.md]: ok. i have removed the readme.md file.

File: core-java-modules/core-java-nio/pom.xml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (KevinGilmore) [inline core-java-modules/core-java-nio/pom.xml]: Let's indent the file using spaces rather than tabs.

Comment (anuraggoyal1) [reply core-java-modules/core-java-nio/pom.xml]: fixed.

**Pull request merged**